specs-go/config.go

```diff
@@ -112,12 +112,11 @@ type Platform struct {
 
 // Mount specifies a mount for a container.
 type Mount struct {
-	// Destination is the path where the mount will be placed relative to the container's root.  The path and child directories MUST exist, a runtime MUST NOT create directories automatically to a mount point.
+	// Destination is the absolute path where the mount will be placed in the container.
 	Destination string `json:"destination"`
 	// Type specifies the mount kind.
-	Type string `json:"type,omitempty"`
-	// Source specifies the source path of the mount.  In the case of bind mounts on
-	// Linux based systems this would be the file on the host.
+	Type string `json:"type,omitempty" platform:"linux,solaris"`
```

Comment:
> I'm not sure I understand this comment.

I'm saying I'd like something closer to:

```Markdown
## Mounts

…current section contents without `type`…

### Linux and Solaris Mounts

For Linux- and Solaris-based systems the mount structure has the following additional properties:

* **`type`** (int, OPTIONAL) …
```

Then a Windows config setting `mounts[].type` would be in the same boat as a Windows config setting `process.users.uid` (although the discussion in #680 shows that what that boat means could still use clarification).